File: backend/app/schemas/producto.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class ProductoBase(BaseModel):
    nombre: str = Field(min_length=1, max_length=150)
    descripcion: str | None = Field(default=None, max_length=500)
    precio: Decimal = Field(gt=0, max_digits=10, decimal_places=2)
    id_categoria: int
    disponible: bool = True
    imagen_url: str | None = Field(default=None, max_length=500)


class ProductoCreate(ProductoBase):
    pass


class ProductoUpdate(BaseModel):
    nombre: str | None = Field(default=None, min_length=1, max_length=150)
    descripcion: str | None = Field(default=None, max_length=500)
    precio: Decimal | None = Field(default=None, gt=0, max_digits=10, decimal_places=2)
    id_categoria: int | None = None
    disponible: bool | None = None
    imagen_url: str | None = Field(default=None, max_length=500)


class ProductoOut(ProductoBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/producto.ts
export interface Producto {
  id: number;
  nombre: string;
  descripcion: string | null;
  precio: string;
  id_categoria: number;
  disponible: boolean;
  imagen_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProductoCreate {
  nombre: string;
  descripcion?: string | null;
  precio: number | string;
  id_categoria: number;
  disponible?: boolean;
  imagen_url?: string | null;
}

export interface ProductoUpdate {
  nombre?: string;
  descripcion?: string | null;
  precio?: number | string;
  id_categoria?: number;
  disponible?: boolean;
  imagen_url?: string | null;
}
